src/main/webapp/resources/scripts/controllers/RootController.coffee

```diff
@@ -8,7 +8,13 @@ angular.module('calendar').controller 'RootController', ($rootScope, $scope, Eve
       event:
         color: '#000000'
 
+  eventCache = null
   $scope.collapseForm = ->
+    index = $rootScope.events.indexOf($scope.form.event)
+    if index != -1
+      $rootScope.events[index] = angular.copy(eventCache)
+    eventCache = null
```

Comment:
Your idea makes new troubles with refreshing view with temporary changes. I've resolved this problem easier way